pkg/agent/controller/endpoint.go

```diff
@@ -140,26 +158,31 @@ func (e *EndpointController) endpointsToEndpointSlice(obj runtime.Object, numReq
 func (e *EndpointController) endpointSliceFromEndpoints(endpoints *corev1.Endpoints, op syncer.Operation) (
 	runtime.Object, bool,
 ) {
-	endpointSlice := &discovery.EndpointSlice{}
-
-	endpointSlice.Name = e.endpointSliceNameFrom(endpoints)
-	endpointSlice.Labels = map[string]string{
-		discovery.LabelManagedBy:        constants.LabelValueManagedBy,
-		constants.LabelSourceNamespace:  e.serviceImportSourceNameSpace,
-		constants.MCSLabelSourceCluster: e.clusterID,
-		mcsv1a1.LabelServiceName:        e.serviceName,
+	endpointSlice := &discovery.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: e.endpointSliceNameFrom(endpoints),
```

Comment:
will the legacy endpoints would be replaced with this new ones ? since name is the same?

Reply:
Yes.